src/clients/SpokePoolClient/SpokePoolClientManager.ts

```diff
@@ -0,0 +1,36 @@
+import winston from "winston";
+import { SpokePoolClient } from "./SpokePoolClient";
+
+/**
+ * SpokePoolClientManager is a wrapper around spokePoolClients. We want to use wrapper almost always
+ * instead of direct access to spokePoolClients because chainId can be invalid and we want to return undefined.
+ */
+export class SpokePoolClientManager {
```

Comment:
I wasn't sure about the naming, but `SpokePoolClientManager` turns out to be a mouthful. wdyt about `SpokePoolManager` instead?

```suggestion
export class SpokePoolManager {
```